extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleApplyRowFilterAndDataMasking.scala

```diff
@@ -30,29 +30,37 @@ import org.apache.kyuubi.plugin.spark.authz.util.RowFilterAndDataMaskingMarker
 class RuleApplyRowFilterAndDataMasking(spark: SparkSession) extends Rule[LogicalPlan] {
 
   override def apply(plan: LogicalPlan): LogicalPlan = {
-    // Apply FilterAndMasking and wrap HiveTableRelation/LogicalRelation with
+    // Apply FilterAndMasking and wrap HiveTableRelation/LogicalRelation/DataSourceV2Relation with
     // RowFilterAndDataMaskingMarker if it is not wrapped yet.
     plan mapChildren {
       case p: RowFilterAndDataMaskingMarker => p
       case hiveTableRelation if hasResolvedHiveTable(hiveTableRelation) =>
         val table = getHiveTable(hiveTableRelation)
-        applyFilterAndMasking(hiveTableRelation, table, spark)
+        applyFilterAndMasking(hiveTableRelation, table.identifier, spark)
       case logicalRelation if hasResolvedDatasourceTable(logicalRelation) =>
         val table = getDatasourceTable(logicalRelation)
         if (table.isEmpty) {
           logicalRelation
         } else {
-          applyFilterAndMasking(logicalRelation, table.get, spark)
+          applyFilterAndMasking(logicalRelation, table.get.identifier, spark)
+        }
+      case datasourceV2Relation if hasResolvedDatasourceV2Table(datasourceV2Relation) =>
```

Comment:
```suggestion
      case relation @ DataSourceV2Relation(_, _, _, Some(identifier), _) if relation.resolved =>
```

Reply:
Good suggestion though. I am still using the `hasXXXX` pattern method to follow the existed method for HiveTableRelation and LogicalRelation as hasResolvedHiveTable and hasResolvedDatasourceTable. And I couldn't see any clarification improvement by using guard pattern for this.

Reply:
May I have your suggestion here. If you voted in @yikf ‘s guard condition pattern suggestion, i will follow it. @yaooqinn

Reply:
let's keep it for multiple spark version support

Reply:
OK. It has bee refactored without importing DataSourceV2Relation.